docs/DeveloperGuide.md

```diff
@@ -8,8 +8,189 @@
+which serves as myManager for the UserUniversityListManager class. UserStorageParser will also create a new TimetableManager,
+and add lessons to the TimetableManager, using the String extracted from the text file.
+
+Relevant exceptions are thrown when there are unexpected scenarios. For example, if the data in the text file is stored in an invalid format,
+an InvalidUserStorageFileException will be thrown.
+
+During the duration of the program, whenever the user decides to alter the data corresponding to UserUniversityListManager
+(ie. add / delete universities or modules, or create new university list), UserStorageParser class will update "data/uni_info.txt" accordingly.
+Similarly, whenever the user decides to alter the data corresponding to TimetableManager (ie. add / delete lessons), UserStorageParser class will update
+"data/timetable_info.txt" accordingly. This is achieved by converting UserUniversityListManager / TimetableManager into a String,
+before saving it in the text file.
+
+The following diagram illustrates the relationships between the two main user storage classes - UserStorage and UserStorageParser.
+
+![User Storage Class Diagram](./images/UserStorage_Class.png)
+
+The following diagram illustrates the flow of the program, from initialisation to updating of the text file.
+
+![User Storage Sequence Diagram](./images/UserStorage_Sequence.png)
+
+### Timetable
+
+Upon starting easySEP, a TimetableManager is created in preparation for users to begin adding Timetables. The TimetableManager is used to create new Timetables, manage existing Timetables and delete old Timetables.
+These Timetables are stored in a HashMap and indexed by university name to facilitate easy reference.
+
+For existing Timetables, they can be used to add and delete Lessons. Each Timetable is made up of a HashMap of ArrayLists(type: Lesson) indexed by weekday e.g. Monday for easy compartmentalisation.
+Within Timetables, Lessons will compromise details like the day, start time and end time. The ArrayLists(type: Lesson) are sorted in non-descending order by the start time of the Lessons.
+
+Relevant exceptions are thrown when unanticipated scenarios occur. For instance, if the user attempts to delete a non-existent Timetable, a TimetableNotFoundException is thrown.
+Another example is the user attempting to add a lesson that is conflicting with an existing lesson in his/her timetable. In this case, a TimetableClashException is thrown.
+
+The following diagram illustrates the relationships between the three main timetable classes - TimetableManager, Timetable and Lesson.
+
+![Timetable Class Diagram](./images/Timetable_Class.png)
+
+In chronological order, the following diagrams illustrate the flow of the program for adding lessons, deleting lessons and displaying timetables to the user.
+
+![Timetable Add Lesson Sequence Diagram](./images/Timetable_addLesson_Sequence.png)
```

Comment:
Detailed and clear diagram. Perhaps might want to put the `isValidDay()`, `isValidStartTime()` etc in a combined `ref` rather than showing all on the diagram as these are trivial functions.